docs/git.rst

```diff
@@ -0,0 +1,611 @@
+
+Not only that,
+it also allows to keep modifying your changes until you and your peers are fine with them.
+
+Further documentation:
+`Introduction to branching <http://git-scm.com/book/en/v2/Git-Branching-Branches-in-a-Nutshell>`_.
+
+
+Commands
+--------
+Some of the most useful/common commands
+(note that most of them have switches that enhance/*completely twist* their functionality):
+
+Just append ``--help`` on all of them to get their full definitions and options,
+i.e. ``git add --help``.
+
+clone
+   Download a repository from a given remote URL.
+
+add
+   Add the given files to the index.
+
+   .. note::
+      **pro tip**: once a file is add via ``git add`` your changes will never be lost!
+      As long as you don't remove the ``.git`` folder,
+      even if you remove the file you just added,
+      the changes you made before doing ``git add`` are still there ready to be recovered at any time!
+
+status
+   Get an overview of the repository status.
+
+   If there are files on the index,
+   or files not tracked by git,
+   or the status of your branch in regards of the remote one,
+   etc.
+
+diff
+   See the current changes made to the files already tracked by git.
+
+   .. note::
+      Fear not, if you used ``git add SOME_FILE`` and then ``git diff`` doesn't output anything you haven't lost your changes!
+
+      Just try ``git diff --cached``.
+      Now you know how to see the working tree changes (``git diff``) and index changes (``git diff --cached``).
+
+commit
+   Create/record changes to the repository
+   (locally only, nothing is sent over the wire).
+
+push
+   Send your changes,
+   either commits or a complete new branch,
+   to the configured remote repository.
+
+show
+   Display the given commit(s) details.
+
+log
+   Shows the repository history.
+   Sorted by date (last commit at the top),
+   and like all other commands,
+   extremely versatile with all its switches.
+
+   See further below for an example of a powerful combination of switches.
+
+branch
+   Create a branch.
+
+fetch
+   Download changes from the remote repository.
+
+   **Without** changing the current ``HEAD`` (see rebase and pull commands).
+
+pull
+   Fetch and integrate changes from remote repository.
+
+   Internally that means to do a ``git fetch`` plus either ``git merge`` or ``git rebase``.
+
+   .. note::
+      Used careless most probably adds extra superfluous commits.
+      See further down.
+
+merge
+   Join two,
+   or more,
+   branches together.
+
+rebase
+   Forward-port your current local commits (or branch) to be based on top of another commit.
+
+   An image is worth 1000 words: http://git-scm.com/docs/git-rebase
+
+checkout
+   Change to the given branch or get the given file to its latest committed version.
+
+   .. note::
+      If git is criticized for being complex,
+      this command is one of the main sources of complains.
+
+      It is so *powerful* yet apparently so *random* that makes people crazy.
```

Comment:
I'm not entirely clear on this sentence. There's nothing really random about this command. It might be helpful to compare this command to svn's 'switch'.